week3/homework/src/actions.js

```diff
@@ -0,0 +1,100 @@
+let fs = require('fs');
+
+function readTodo(request, response) {
+  fs.readFile('./todoList.json', 'utf8', (error, todoList) => {
+    const parsedList = JSON.parse(todoList);
+    if (error) {
+      if (error.code === 'EN0ENT') {
+        console.log('No data found');
+      } else {
+        console.error(error);
+        response.status(500).send('some problem.');
+      }
+    } else if (request.params.id < parsedList.length) {
+      const todoItem = parsedList[++request.params.id]['text'];
+      response.json(todoItem);
+    } else {
+      response.json('Please enter a valid id');
+    }
+  });
+}
+
+function add(request, response) {
+  fs.readFile('./todoList.json', 'utf8', (error, data) => {
+    if (error) {
+      console.log(error);
+    } else {
+      let todos = JSON.parse(data);
+      const todo = { text: request.params.text, done: false };
+      todos.push(todo);
+      let dataString = JSON.stringify(todos);
+      fs.writeFileSync('./todoList.json', dataString, 'utf8');
```

Comment:
You need to handle the error it might happens on file-write, and send the proper status code accordingly.